src/pdf_to_json/pdf_to_civiform_gemini.py

```diff
@@ -120,44 +180,102 @@ def process_pdf_text_with_llm(client, model_name, file, base_name):
+        page_count = get_pdf_page_count(file)
+        logging.info(f"Page count {page_count}")
+        responses = []
 
         if not model_name.startswith("models/"):
             api_model_name = f"models/{model_name}"
         else:
             api_model_name = model_name
 
-        # This specific call structure might depend on the exact library version
-        # It accesses the 'models' attribute of the client, then calls generate_content
+        input_file = types.Part.from_bytes(data=file, mime_type="application/pdf")
         response = client.models.generate_content(
-             model=api_model_name, # Use potentially prefixed model name
-             contents=[input_part, prompt]
-             # TODO add safety_settings here
+            model=api_model_name,
+            contents=[input_file, prompt]
         )
 
-        # add robust check based on actual library behavior
+        # Add robust check based on actual library behavior
         if hasattr(response, 'text'):
-             response_text = response.text.strip("`").lstrip(
-                 "json")  # Remove ``` and "json" if present
-        elif hasattr(response, 'candidates') and response.candidates and hasattr(response.candidates[0], 'content') and response.candidates[0].content.parts:
-             # Fallback to candidate structure if .text isn't available (more typical)
-             response_text = response.candidates[0].content.parts[0].text.strip("`").lstrip("json").strip()
+            response_text = response.text.strip("`").lstrip("json")  # Remove ``` and "json" if present
+        elif (
+            hasattr(response, 'candidates') and response.candidates and
+            hasattr(response.candidates[0], 'content') and response.candidates[0].content.parts
+        ):
+            # Fallback to candidate structure if .text isn't available (more typical)
+            response_text = (
+                response.candidates[0].content.parts[0].text
+                .strip("`")
+                .lstrip("json")
+                .strip()
+            )
         else:
-             # Cannot extract text, log the response structure for debugging
-             logging.error(f"Could not extract text from LLM response. Response object: {response}")
-             return None, "Failed to extract text from LLM response"
-
-
-        if logging.getLogger().getEffectiveLevel() <= logging.DEBUG:
-            # Uses the global output_json_dir implicitly
-            save_response_to_file(
-                response_text, base_name, f"pdf-extract-{model_name}", output_json_dir)
+            # Cannot extract text, log the response structure for debugging
+            logging.error(f"Could not extract text from LLM response. Response object: {response}")
+            return None, "Failed to extract text from LLM response"
 
-        return response_text, None  # Return response and None for error
+        try:
+          json_response = json.loads(response_text.strip())
```

Comment:
@orwant This is the logic where I am trying to process the entire pdf - if there is a failure L224 - then process in batches. The `fix_malformed_json` function. is like a sanity check - L261 to ensure the entire json is correct before post processing. 

Post processing divides the big json into chunks (sections)-> runs the post processing prompt  and merges it back into the final json.

The splitting is by pages at this time - we can split by sections in the next iteration (involves adding another prompt)